tests/test_decl.py

```diff
@@ -206,6 +206,87 @@ def test_decl_can_create_simple_objects_by_passing_plain_strings(
 
         assert "NewGame++" in attrdef.values
 
+    @staticmethod
+    def test_extend_operations_change_model_objects_in_place(
+        model: capellambse.MelodyModel,
+    ) -> None:
+        root_function = model.by_uuid(ROOT_FUNCTION)
+        fnc = model.by_uuid("8833d2dc-b862-4a50-b26c-6f7e0f17faef")
+        old_parent = fnc.parent
+
+        assert fnc not in root_function.functions, "Precondition failed"
+
+        yml = f"""\
+            - parent: !uuid {ROOT_FUNCTION}
+              extend:
+                functions:
+                  - !uuid {fnc.uuid}
+            """
+
+        decl.apply(model, io.StringIO(yml))
+
+        assert fnc in root_function.functions
+        assert fnc.parent == root_function != old_parent
+        assert fnc not in old_parent.functions  # type: ignore[operator]
+
+    @staticmethod
+    def test_extend_operations_with_promises_change_model_objects_in_place(
+        model: capellambse.MelodyModel,
+    ) -> None:
+        root_component = model.by_uuid(ROOT_COMPONENT)
+        fnc_name = "The promised one"
+
+        assert fnc_name not in model.la.all_functions.by_name
+        assert fnc_name not in root_component.functions.by_name
+
+        yml = f"""\
+            - parent: !uuid {ROOT_FUNCTION}
+              extend:
+                functions:
+                  - name: {fnc_name}
+                    promise_id: promised-fnc
+            - parent: !uuid {ROOT_COMPONENT}
+              extend:
+                functions:
+                  - !promise promised-fnc
+            """
+
+        decl.apply(model, io.StringIO(yml))
+        fnc_from_lookup = model.la.all_functions.by_name(fnc_name)
+        fnc_from_comp_functions = root_component.functions.by_name(fnc_name)
+
+        assert fnc_from_lookup == fnc_from_comp_functions
```

Comment:
While technically mostly correct (I think), this assertion looks quite weird for what it's intended to do.

We should rather obtain one (persistent) reference to the object-under-test (i.e. the `ROOT_COMPONENT`) before executing `decl.apply()`, check as precondition that it does not have a "The promised one" function, and then check as test result that it now has one.